src/main/java/edu/ie3/datamodel/io/sink/SqlSink.java

```diff
@@ -0,0 +1,521 @@
+     * @param entities a collection of entities that should be persisted
+     * @param identifier identifier of the grid
+     */
+    public <C extends InputEntity> void persistAllIgnoreNested(Collection<C> entities, DatabaseIdentifier identifier) {
+        persistMixedList(new ArrayList<>(entities), identifier);
+    }
+
+    /**
+     * Persist an entity. By default this method take care about the extraction process of nested entities (if any)
+     * @param entity the entity that should be persisted
+     * @param identifier identifier of the grid
+     * @throws SQLException
+     */
+    public <C extends UniqueEntity> void persist(C entity, DatabaseIdentifier identifier) throws SQLException {
+        if (entity instanceof InputEntity inputEntity) {
+            persistIncludeNested(inputEntity, identifier);
+        } else if (entity instanceof ResultEntity resultEntity) {
+            insert(resultEntity, identifier);
+        } else if (entity instanceof TimeSeries<?, ?> timeSeries) {
+            persistTimeSeries(timeSeries, identifier);
+        } else {
+            log.error("I don't know how to handle an entity of class {}", entity.getClass().getSimpleName());
+        }
+    }
+
+    /**
+     * Persist an entity. In contrast to {@link SqlSink#persist} this function doe not extract nested entities.
+     * @param entity the entity that should be persisted
+     * @param identifier identifier of the grid
+     * @throws SQLException
+     */
+    public <C extends InputEntity> void persistIgnoreNested(C entity, DatabaseIdentifier identifier) throws SQLException {
+        insert(entity, identifier);
+    }
+
+    // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+    protected <C extends UniqueEntity> void persistListIncludeNested(List<C> entities, Class<C> cls, DatabaseIdentifier identifier) throws SQLException {
+        if (NestedEntity.class.isAssignableFrom(cls)) {
+            entities.forEach(
+                    entity -> {
+                        try {
+                            List<InputEntity> arr = new ArrayList<>(Extractor.extractElements((NestedEntity) entity));
+                            persistAll(arr, identifier);
+                        } catch (ExtractorException | SQLException e) {
+                            log.error(
+                                    String.format(
+                                            "An error occurred during extraction of nested entity'%s': ",
+                                            cls.getSimpleName()),
+                                    e);
+                        }
+                    }
+            );
+            insertListIgnoreNested(entities, cls, identifier);
+        } else {
+            insertListIgnoreNested(entities, cls, identifier);
+        }
+    }
```

Comment:
Why do we need this on top of `persistAll`?

Reply:
Good question, I delete it.